src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1347,10 +1347,14 @@ internal void CompleteCompilationEventQueue_NoLock()
         /// </summary>
         /// <param name="accumulator">Bag to which filtered diagnostics will be added.</param>
         /// <param name="incoming">Diagnostics to be filtered.</param>
+        /// <param name="filterDiagnostics">
+        /// Flag indicating if diagnostics should be filtered or not.
+        /// This flag should always be true, except when this method is invoked in a 
+        /// command line compilation scenario.</param>
```

Comment:
Why do we have this exception and why do we need to mention it here?

Reply:
I thought this would be useful for the callers, I can remove this comment if it seems to add more confusion.

Reply:
> I thought this would be useful for the callers, I can remove this comment if it seems to add more confusion.

It doesn't feel right to me to have any extra exceptions/requirements for a command line build vs. any other build. 

---
In reply to: [289968290](https://github.com/dotnet/roslyn/pull/36067#discussion_r289968290) [](ancestors = 289968290)